src/ints/int10_vesa.cpp

```diff
@@ -129,22 +130,32 @@ Bit8u VESA_GetSVGAModeInformation(Bit16u mode,Bit16u seg,Bit16u off) {
 	MODE_INFO minfo;
 	memset(&minfo,0,sizeof(minfo));
 	PhysPt buf=PhysMake(seg,off);
-	Bitu pageSize;
+	int pageSize = 0;
 	Bit8u modeAttributes;
-	Bitu i=0;
 
 	mode&=0x3fff;	// vbe2 compatible, ignore lfb and keep screen content bits
 	if (mode<0x100) return 0x01;
 	if (svga.accepts_mode) {
 		if (!svga.accepts_mode(mode)) return 0x01;
 	}
-	while (ModeList_VGA[i].mode!=0xffff) {
-		if (mode==ModeList_VGA[i].mode) goto foundit; else i++;
+
+	// Find the requested mode in our table of VGA modes
+	auto mblock = ModeList_VGA.end();
+	for (auto v = ModeList_VGA.begin(); v != ModeList_VGA.end(); ++v) {
```

Comment:
how about `for (auto v: Modelist_VGA)` and using the elements directly instead of via pointers? So the above line would be `auto mblock = Modelist_VGA.back()`, etc.

Reply:
Converted - here's how it changed:
https://github.com/dosbox-staging/dosbox-staging/pull/1244/commits/a0fdd1ef97b48ccbe6bc87d629f22a0faeed1dd0